src/modules/graphics/Shader.cpp

```diff
@@ -671,6 +715,10 @@ bool Shader::validateInternal(ShaderStage *vertex, ShaderStage *pixel, std::stri
 		reflection.usesPointSize = vertintermediate->inIoAccessed("gl_PointSize");
 	}
 
+	if (stages[SHADERSTAGE_COMPUTE] != nullptr)
+		for (int i = 0; i < 3; i++)
+			reflection.localThreadgroupSize[i] = program.getLocalSize(i);
```

Comment:
I don't remember whether glslang / glsl itself will error on its own if `local_size` is omitted or has <= 0 values - I think it's probably worth explicitly outputting an error here if any threadgroup size is <= 0.